lib/charms/opensearch/v0/opensearch_secrets.py

```diff
@@ -113,13 +113,7 @@ def _on_secret_changed(self, event: SecretChangedEvent):  # noqa: C901
 
         logger.debug("Secret change for %s", str(label_key))
 
-        # Leader has to maintain TLS and Dashboards relation credentials
-        if not is_leader and label_key == CertType.APP_ADMIN.val:
-            self._charm.tls.store_new_tls_resources(CertType.APP_ADMIN, event.secret.get_content())
-            if self._charm.tls.is_fully_configured():
-                self._charm.peers_data.put(Scope.UNIT, "tls_configured", True)
-
-        elif is_leader and label_key == self._charm.secrets.password_key(KibanaserverUser):
```

Comment:
Updating the TLS certs after a `secret_change` event has been removed. New TLS certs should only be processed by the `certificate_available` event in order to make sure new CA's are recognized and the workflow for rotating the CA and renewing the certificates is being handled correctly.